python/GafferImageUI/CatalogueUI.py

```diff
@@ -1116,20 +1164,38 @@ def __columnContextMenuDefinition( self ) :
 
 		return menu
 
+	def __outputIndexContextMenuDefinition( self, image ) :
+
+		menu = IECore.MenuDefinition()
+		for i in range( 1, 5 ) :
+
+			menu.append( "/Output Index %i" % i, {
+				"checkBox" : image["outputIndex"].getValue() == i,
+				"command" : functools.partial( Gaffer.WeakMethod( self.__setComparisonImage ), i, image  ),
+			} )
+
+		return menu
+
 	def __contextMenu( self, *unused ) :
 
 		if self.getPlug() is None :
 			return False
 
+
 		# This signal is called anywhere in the listing, check we're over the header.
 		mousePosition = GafferUI.Widget.mousePosition( relativeTo = self.__pathListing )
 		headerRect = self.__pathListing._qtWidget().header().rect()
-		if not headerRect.contains( mousePosition[0], mousePosition[1] ) :
-			return False
-
-		self.__popupMenu = GafferUI.Menu( self.__columnContextMenuDefinition(), "Columns" )
-		self.__popupMenu.popup( parent = self )
+		if headerRect.contains( mousePosition[0], mousePosition[1] ) :
+			self.__popupMenu = GafferUI.Menu( self.__columnContextMenuDefinition(), "Columns" )
+			self.__popupMenu.popup( parent = self )
+			return True
+		elif isinstance( self.__pathListing.columnAt( mousePosition ), OutputIndexColumn ):
+			image = _imageFromName( self.getPlug().node(), self.__pathListing.pathAt( mousePosition )[0] )
+			if image:
+				self.__popupMenu = GafferUI.Menu( self.__outputIndexContextMenuDefinition( image ), "Set As Output Index" )
```

Comment:
`title = "Output Index"`. `"Set as Output Index"` is a positional argument getting swallowed by `_qtParent`.